# ***************************************************************************** #
# PARAMETERS SPECIFIC TO THE REMOTE CONTROL FEATURE

# Setpoint for the HORIZONTAL test
test_horizontal_setpoint1 : [0.3, 0.0, 0.4, 0.0]
test_horizontal_setpoint2 : [-0.3, 0.0, 0.4, 0.0]

# Setpoint for the VERTICAL test
test_vertical_setpoint1 : [0.0, 0.0, 0.4, 0.0]
test_vertical_setpoint2 : [0.0, 0.0, 0.7, 0.0]

# Setpoint for the HEADING test
test_heading_setpoint1 : [0.0, 0.0, 0.4, 0.0]
test_heading_setpoint2 : [0.0, 0.0, 0.4, 3.14]

# Setpoint for the ALL test
test_all_setpoint1 : [0.3, 0.0, 0.4, 0.0]
test_all_setpoint2 : [-0.3, 0.0, 0.7, 1.57]

# Parameters for flying in a circle
test_circle_radius : 0.40
test_circle_seconds_per_rev : 3.0
test_circle_height : 0.4
test_circle_pirouette_per_rev : 0.0
test_circle_time_to_reach_start : 0.7

# Multipliers for the HORIZONTAL contorller
multiplier_horizontal_min : 0.1
multiplier_horizontal_max : 2.2
# Multipliers for the VERTICAL contorller
multiplier_vertical_min : 0.1
multiplier_vertical_max : 1.7
# Multipliers for the HEADING contorller
multiplier_heading_min : 0.05
multiplier_heading_max : 3.0


# ***************************************************************************** #



# Mass of the crazyflie
mass : 30

# Frequency of the controller, in hertz
vicon_frequency : 200
control_frequency : 200

# Quadratic motor regression equation (a0, a1, a2)
motorPoly : [5.484560e-4, 1.032633e-6, 2.130295e-11]

# Boolean for whether to execute the convert into body frame function
shouldPerformConvertIntoBodyFrame : true

# Boolean indiciating whether the "Debug Message" of this agent should be published or not
shouldPublishDebugMessage : false

# Boolean indiciating whether the debugging ROS_INFO_STREAM should be displayed or not
shouldDisplayDebugInfo : false


# A flag for which controller mode to use, defined as:
# 1 CONTROLLER_MODE_LQR_MOTOR
#    -  LQR controller based on the state vector: [position,velocity,angles,angular velocity]
#       commands per motor thrusts
# 2 CONTROLLER_MODE_LQR_ACTUATOR
#    -  LQR controller based on the state vector: [position,velocity,angles,angular velocity]
#       commands actuators of total force and bodz torques
# 3 CONTROLLER_MODE_LQR_RATE
#    -  LQR controller based on the state vector: [position,velocity,angles]
# 4 CONTROLLER_MODE_LQR_ANGLE
#    -  LQR controller based on the state vector: [position,velocity]
# 5 CONTROLLER_MODE_LQR_ANGLE_RATE_NESTED
#    -  LQR Nested angle and rate controller
# 6 CONTROLLER_MODE_ANGLE_RESPONSE_TEST
#    -  Swaps between pitch set-points to test angle set-point response time
#       i.e., this controller test the assumption that "the inner loop is infinitely fast"
#
controller_mode : 5


# A flag for which estimator to use, defined as:
# 1  -  Finite Different Method,
#       Takes the poisition and angles directly as measured,
#       and estimates the velocities as a finite different to the
#       previous measurement
# 2  -  Point Mass Per Dimension Method
#       Uses a 2nd order random walk estimator independently for
#       each of (x,y,z,roll,pitch,yaw)
# 3  -  Quad-rotor Model Based Method
#       Uses the model of the quad-rotor and the previous inputs
estimator_method : 1


# The LQR Controller parameters for "mode = 3"
gainMatrixThrust_NineStateVector    :  [ 0.00, 0.00, 0.76, 0.00, 0.00, 0.32, 0.00, 0.00, 0.00]
gainMatrixRollRate                  :  [ 0.00,-1.80, 0.00, 0.00,-1.38, 0.00, 5.20, 0.00, 0.00]
gainMatrixPitchRate                 :  [ 1.80, 0.00, 0.00, 1.38, 0.00, 0.00, 0.00, 5.20, 0.00]
gainMatrixYawRate                   :  [ 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 2.30]

# The LQR Controller parameters for "mode = 4"
gainMatrixThrust_SixStateVector     :  [ 0.00, 0.00, 0.98, 0.00, 0.00, 0.25]
gainMatrixRollAngle                 :  [ 0.00,-0.20, 0.00, 0.00,-0.20, 0.00]
gainMatrixPitchAngle                :  [ 0.20, 0.00, 0.00, 0.20, 0.00, 0.00]

# The LQR Controller parameters for "mode = 5"
gainMatrixThrust_SixStateVector_50Hz:  [ 0.00, 0.00, 0.82, 0.00, 0.00, 0.22]
gainMatrixRollAngle_50Hz            :  [ 0.00,-0.31, 0.00, 0.00,-0.25, 0.00]
gainMatrixPitchAngle_50Hz           :  [ 0.31, 0.00, 0.00, 0.25, 0.00, 0.00]

gainMatrixRollRate_Nested           :  [ 4.00, 0.00, 0.00]
gainMatrixPitchRate_Nested          :  [ 0.00, 4.00, 0.00]
gainMatrixYawRate_Nested            :  [ 0.00, 0.00, 2.30]


# The max and minimum thrust for a 16-bit command
command_sixteenbit_min : 1000
command_sixteenbit_max : 60000


# THE POINT MASS KALMAN FILTER (PMKF) GAINS AND ERROR EVOLUATION
# > For the (x,y,z) position
PMKF_Ahat_row1_for_positions  :  [  0.6723, 0.0034]
PMKF_Ahat_row2_for_positions  :  [-12.9648, 0.9352]
PMKF_Kinf_for_positions       :  [  0.3277,12.9648]


# > For the (roll,pitch,yaw) angles
PMKF_Ahat_row1_for_angles     :  [  0.6954, 0.0035]
PMKF_Ahat_row2_for_angles     :  [-11.0342, 0.9448]
PMKF_Kinf_for_angles          :  [  0.3046,11.0342]
